src/main/scala/PrivateHome/data.scala

```diff
@@ -1,47 +1,180 @@
 import PrivateHome.Devices.Switch
+import scalikejdbc._
 
-import scala.xml._
 
 object data {
+  /**
+   * An map containing all settings
+   */
+  val settings: Map[String, Int] = Map(("port", 2888), ("ip", 565))
+
+
+  GlobalSettings.loggingSQLAndTime = LoggingSQLAndTimeSettings(
+    enabled = false,
+    singleLineMode = false,
+    printUnprocessedStackTrace = false,
+    stackTraceDepth = 15,
+    logLevel = Symbol("debug"),
+    warningEnabled = true,
+    warningThresholdMillis = 3000L,
+    warningLogLevel = Symbol("warn")
+  )
+
+
+  Class.forName("org.h2.Driver")
+  ConnectionPool.singleton("jdbc:h2:./daten/Devices", "user", "pass")
+
+  implicit val session: AutoSession.type = AutoSession
+  var devices: Map[String, Switch] = Map()
+  var mhzId: Map[String, String] = Map()
 
   /**
-   * This variable returns a List containing all ids.
+   * Generates the Table strucktures for the Database
    */
-  val IDs: List[String] = {
-    val xml = XML.load("src/main/scala/PrivateHome/id.xml")
-    val prep: NodeSeq = xml \\ "id"
-    val result = for (i <- 0 until prep.length) yield prep(i).text
-    result.toList
+  def create(): Unit = {
+
+    // Drops old Tables to Delet Data and make it possible to regenerate them
+    sql"""DROP TABLE IF EXISTS `Mhz`;
+         DROP TABLE IF EXISTS `Devices`;""".execute().apply()
+
+
+    val devices =
+      sql"""
+                    CREATE TABLE `Devices` (
+                    `id` varchar(5) NOT NULL,
+                    `name` varchar(64) NOT NULL,
+                    `type` varchar(16) NOT NULL,
+                    `state` decimal(5,4) NOT NULL,
+                    `keepState` boolean NOT NULL,
+                    PRIMARY KEY (`id`))
+         """.execute.apply()
+
+
+    sql"""
+           CREATE TABLE `Mhz` (
+           `id` varchar(5) NOT NULL,
+           `systemcode` varchar(5) NOT NULL,
+           `unitcode` varchar(5) NOT NULL,
+           PRIMARY KEY (`id`),
+           CONSTRAINT `Mhz_ibfk_1` FOREIGN KEY (`id`) REFERENCES `Devices` (`id`))
+         """.execute().apply()
+
+
+    // insert initial data
+
+
   }
 
   /**
-   * This variable returns a map which contains the id as the key and the Switch-object for this id.
+   * Fills the devices Map with all Devices from the DB and turns all Switches with keepState on
    */
-  val devices: Map[String, Switch] = {
-    val xml: Elem = XML.load("src/main/scala/PrivateHome/devices.xml")
-    val prep: NodeSeq = xml \ "switch"
-    val data: Seq[(String, Switch)] = for (i <- prep) yield (i \ "@id").text -> Switch(i)
-    data.toMap
+  def fillDevices(): Unit = {
+    val m = Device.syntax("m")
+    withSQL {
+      select.from(Device as m)
+    }.map(rs => Device(rs)).list().apply().foreach(d => {
+      d.switchtype match {
+        case "MQTT" => devices += ((d.id, mqttSwitch(d.id, keepStatus = d.keepState)))
+          if (d.keepState) {
+            devices(d.id).on(d.state)
+          }
```

Comment:
This is all it needs for MQTT support because it creates the Switch and appends it to the devices map

Reply:
ok